### DISCUSSION TITLE
Create new vesting transaction to represent accelerated vesting

### DISCUSSION THREAD

**Discussion by pjohnmeyer** [Open-Cap-Table-Coalition/Open-Cap-Format-OCF#271] [Ideas]
## Why

Currently, there is no way to represent a vesting acceleration in the transaction log.

**Note:** This is not meant to replace milestone vesting via [VestingEvent](https://github.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/blob/main/schema/objects/transactions/vesting/VestingEvent.schema.json). Those events reference a milestone that are part of the vesting terms, whereas acceleration is an "advance" on shares that are scheduled to be vested in the future.

## Proposal

Create a new event:

```json
{
  "object_type": "TX_VESTING_ACCELERATE", // required, const
  "reason_text": "some platform provided reason, not enumerated", // optional, string
  "quantity": "100", // required, numeric
  "date": "2022-02-02", // required, date
  ... remainder of Object / Transaction / SecurityTransaction fields
}
```

**Question** Do we need an optional "effective date" or "board approval date" on this as well?

### Do nothing
If we do not create a VestingAccelerate event, then any security that has vesting accelerated will need to create a new [VestingTerms](https://github.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/blob/v1.0.0-b2/schema/objects/VestingTerms.schema.json) object unique to that security. While this is a valid alternative, acceleration is a common enough occurrence that I believe the schema should natively support it.

### Represent vesting acceleration as an event in the VestingTerms
We could add a new trigger and represent accelerated vesting as an event-based [VestingCondition](https://github.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/blob/v1.0.0-b2/schema/types/vesting/VestingCondition.schema.json). Doing this, however, would greatly complicate the graph, and would require more parameterization of the condition. This has similar complexities expressed in #269 that I won't repeat here.

**Comment by JSv4** [comment]
@pjohnmeyer, good catch. As I recall, we thought we could get away with your "do nothing" option, but I agree that having to create a new VestingTerms object for every applicable issuance is a pain. Definitely needed if not part of the spec. Open to discussing at today's TWG?

**Comment by EricVogl-carta** [comment]
We would likely also want to specify "which" shares get accelerated. Not all accelerations are the same, most are LIFO (last to vest, after considering partial cancels). Some could be accelerating specific shares from specific tranches. Unsure if this level of granularity would be needed generally or if we defer the second scenario to custom VestingTerms.

**Reply by roehst** [reply]
At work we roughly follow the format in our implementation, and we leverage event vestings for that. All of our vesting conditions can have both a date trigger and an event trigger which we name “Acceleration”.
